pkg/command/context.go

```diff
@@ -2143,14 +2142,30 @@ func renderDynamicTable(slices interface{}, tableWriter component.OutputWriter,
 }
 
 func mapTanzuEndpointToTMCEndpoint(tanzuEndpoint string) string {
-	// Define the regular expression pattern
-	apiPattern := regexp.MustCompile(`https://api\.tanzu(-\w*)?\.cloud\.vmware\.com`)
-	// Replace "api" with "tmc" in the input URL
-	tmcEndpoint := apiPattern.ReplaceAllString(tanzuEndpoint, "https://tmc.tanzu$1.cloud.vmware.com")
+	tmcEndpoint := ""
+	// Define the mapping rules
+	mappingRules := []struct {
+		prefix string
+		suffix string
+	}{
+		{"https://api.tanzu", "https://tmc.tanzu"},
```

Comment:
wondering: should we be concerned that we are replacing for not just .cloud.vmware.com/.tanzu.broad.com but _every_ other domain?

Reply:
Yeah. I wanted to keep things flexible as much as possible for now. So this is intentional. Do you think we need specific one?

Reply:
We had an offline discussion and decided to keep this flexible as we have in this change.